test/test_broker.hpp

```diff
@@ -810,15 +840,25 @@ class test_broker {
         auto send_offline_messages =
             [&] (session_state& session) {
                 try {
-                    while (!session.offline_messages.empty()) {
-                        auto const& msg = session.offline_messages.front();
-                        session.con->publish(
-                            msg.topic,
-                            msg.contents,
-                            msg.pubopts,
-                            msg.props
-                        );
-                        session.offline_messages.pop_front();
+                    auto &seq_idx = session.offline_messages.get<tag_seq>();
+                    while(!seq_idx.empty()) {
+                        seq_idx.modify(seq_idx.begin(), [&](auto &i) {
+                            auto props = MQTT_NS::force_move(i.props);
+
+                            if(i.tim_message_expiry) {
+                                set_property<MQTT_NS::v5::property::message_expiry_interval>(props,
+                                    MQTT_NS::v5::property::message_expiry_interval(static_cast<uint32_t>(std::chrono::duration_cast<std::chrono::seconds>(i.tim_message_expiry->expires_from_now()).count())));
```

Comment:
expires_from_now() is deprecated.
See https://www.boost.org/doc/libs/1_74_0/doc/html/boost_asio/reference/steady_timer.html

Try compile with `-DBOOST_ASIO_NO_DEPRECATED`.